js/apps/admin-ui/src/App.tsx

```diff
@@ -20,21 +20,28 @@ import { SubGroups } from "./groups/SubGroupsContext";
 import { AuthWall } from "./root/AuthWall";
 
 const AppContexts = ({ children }: PropsWithChildren) => (
-  <RealmsProvider>
-    <RealmContextProvider>
-      <WhoAmIContextProvider>
-        <RecentRealmsProvider>
-          <AccessContextProvider>
-            <Help>
-              <AlertProvider>
-                <SubGroups>{children}</SubGroups>
-              </AlertProvider>
-            </Help>
-          </AccessContextProvider>
-        </RecentRealmsProvider>
-      </WhoAmIContextProvider>
-    </RealmContextProvider>
-  </RealmsProvider>
+  <ErrorBoundary
```

Comment:
What is the reason that the `ErrorBoundary` was moved up in the chain? This causes the error message to render as the only thing on the page, where before this was rendered inside of the `Page`.

Reply:
This was the main caveat about this version upgrade

In the new react-error-boundary version, as far as I understand what's happening under the hood, the new  `useErrorBoundary` needs a context during it's initialization. when the previous `useErrorHandler` didn't in the v3 (I mean, it also needed a ErrorBoundary component wrapper but the initialisation of the component didn't throw an error if none was found in the parents.

And the RealmsProvider component actually use the useFetch hook 

So it was fine with the v3 even if the RealmsProvider was outside the ErrorBoundary component. (I must admit, I couldn't tell it the react-error-boundary behaviour was ok in case of error in the RealmsProvider)

with the v4 though, the context is not yet initialized during the rendering of `RealmsProvider` and so the `useErrorBoundary` called in `useFetch` fails

The simplest fix was to wrap also the RealmsProvider (and any component needing `useFetch` hook) in the  `ErrorBoundary` Component.  Which actually changes the rendering of errors :/

I guess another way would be to init the context of react-error-boundary independently (may not be natively handled by the lib) but I'm not sure about how a RealmsProvider error should be rendered then.

Or maybe we could use multiple ErrorBoundary element to handle errors during the RealmsProvider rendering in a different way than for the rest of the UI (which would stick with the old way then)

But I'm afraid that"s not a simple version upgrade anymore in this case.

Reply:
Hmmm yeah, that complicates things a lot, and is actually quite a silly change on the side of `react-error-boundary`, let me do some digging to understand this better.

Reply:
You are correct, it is not possible to implement the same logic using the new version. Since we only use a subset of the functionality I have created #26094 that rolls a custom implementation, let me know what you think.